ukis_pysat/raster.py

```diff
@@ -194,26 +196,21 @@ def warp(self, dst_crs, resampling_method=0, num_threads=4, resolution=None):
 
         destination = np.zeros((self.dataset.count, height, width), self.__arr.dtype)
 
-        for i in range(0, self.dataset.count):
-            reproject(
-                source=rasterio.band(self.dataset, i + 1),  # index starting at 1
-                destination=destination[i],
-                src_transform=self.dataset.transform,
-                src_crs=self.dataset.crs,
-                dst_transform=transform,
-                dst_crs=dst_crs,
-                resampling=resampling_method,
-                num_threads=num_threads,
-            )
-
-        # update for further processing
-        self.__arr = destination
-        self.transform = transform
-        self.crs = dst_crs
+        self.__arr, transform = reproject(
+            source=self.__arr,
+            destination=destination,
+            src_transform=self.dataset.transform,
+            src_crs=self.dataset.crs,
+            src_nodata=nodata,
+            dst_transform=transform,
+            dst_crs=dst_crs,
+            dst_nodata=nodata,
+            resampling=resampling_method,
+            num_threads=num_threads,
+        )
 
         self.dataset.close()

```

Comment:
Do we still need to close the dataset at this stage? If yes please do so consistently in all methods (e.g. in dn2toa you do not close the dataset before update).

Reply:
I agree, just forgot it there.